### ISSUE TITLE
panic: runtime error: index out of range

### ISSUE THREAD

**Issue by princemaple** [ehazlett/interlock#119]
while it works perfectly on local boot2docker machine created by docker machine

it gives me this

```
 ❯ docker-compose up interlock
Creating network "app_default" with the default driver
Creating volume "app_pg_data" with local driver
Creating app_interlock_1
Attaching to app_interlock_1
interlock_1 | time="2016-04-13T07:27:25Z" level=info msg="interlock 1.1.0 (8a68c99)"
interlock_1 | time="2016-04-13T07:27:25Z" level=debug msg="loading config from: /bin/config.toml"
interlock_1 | time="2016-04-13T07:27:25Z" level=debug msg="docker client: url=unix:///var/run/docker.sock"
interlock_1 | time="2016-04-13T07:27:25Z" level=debug msg="loading extension: name=haproxy"
interlock_1 | panic: runtime error: index out of range
interlock_1 |
interlock_1 | goroutine 1 [running]:
interlock_1 | panic(0xba2240, 0xc820010010)
interlock_1 |   /usr/local/go/src/runtime/panic.go:464 +0x3e6
interlock_1 | github.com/ehazlett/interlock/ext/lb.getNodeID(0x0, 0x0, 0x0, 0x0)
interlock_1 |   /go/src/github.com/ehazlett/interlock/ext/lb/utils.go:35 +0x52d
interlock_1 | github.com/ehazlett/interlock/ext/lb.NewLoadBalancer(0xc820170480, 0xc82016d920, 0xcd5798, 0x0, 0x0)
interlock_1 |   /go/src/github.com/ehazlett/interlock/ext/lb/lb.go:93 +0x1d0
interlock_1 | github.com/ehazlett/interlock/server.(*Server).loadExtensions(0xc82016d800, 0xc82016d920)
interlock_1 |   /go/src/github.com/ehazlett/interlock/server/server.go:171 +0x236
interlock_1 | github.com/ehazlett/interlock/server.NewServer(0xc820112b00, 0x16d, 0x0, 0x0)
interlock_1 |   /go/src/github.com/ehazlett/interlock/server/server.go:112 +0x280
interlock_1 | main.runAction(0xc820137040)
interlock_1 |   /go/src/github.com/ehazlett/interlock/cmd/interlock/run.go:186 +0xc8f
interlock_1 | github.com/ehazlett/interlock/vendor/github.com/codegangsta/cli.Command.Run(0xcd7fd0, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0xcf0ae0, 0xd, 0x0, ...)
interlock_1 |   /go/src/github.com/ehazlett/interlock/vendor/github.com/codegangsta/cli/command.go:174 +0x1399
interlock_1 | github.com/ehazlett/interlock/vendor/github.com/codegangsta/cli.(*App).Run(0xc820136dc0, 0xc82000a0a0, 0x5, 0x5, 0x0, 0x0)
interlock_1 |   /go/src/github.com/ehazlett/interlock/vendor/github.com/codegangsta/cli/app.go:187 +0x112b
interlock_1 | main.main()
interlock_1 |   /go/src/github.com/ehazlett/interlock/cmd/interlock/main.go:36 +0x3db
app_interlock_1 exited with code 2
```

on 

```
 ❯ uname -rsm
Linux 4.4.4-std-3 x86_64
```

Would you happen to know why?

I tried pulling this repo and running the examples, same error.

**EDIT:** changed `dc` to `docker-compose` to reflect that it's just an alias

**Comment by princemaple** [comment]
FYI this is on a VPS on scaleway.com

**Comment by ehazlett** [comment]
Ya Interlock needs to run in a container. That's the getNodeID error
although it should report a better error. I will take a look.

**Comment by princemaple** [comment]
That's weird, it **is** run in a container. Hmm. Is there any other information I can provide you with?

**Comment by princemaple** [comment]
And thank you so much for the super timely response.

**Comment by ehazlett** [comment]
Oh odd. Can you give me some more info about the host?  Docker version, OS
etc.

**Comment by ehazlett** [comment]
Yw :). It's late for me so I will check this out more in the morning. :).
Host info will help. I'll try to reproduce. Thx!

**Comment by princemaple** [comment]
```
 ⚡ root@scw-10c99b  /v/app 
 ❯ docker-compose version
docker-compose version 1.6.2, build 4d72027
docker-py version: 1.7.2
CPython version: 2.7.9
OpenSSL version: OpenSSL 1.0.1e 11 Feb 2013

 ⚡ root@scw-10c99b  /v/app 
 ❯ uname -a
Linux scw-10c99b 4.4.4-std-3 #1 SMP Tue Mar 8 17:31:34 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

my compose file

```yml
version: '2'
services:
  interlock:
    image: ehazlett/interlock:1.1.0
    command: run -c /bin/config.toml
    ports:
      - 8080
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - ./config.toml:/bin/config.toml
  lb:
    image: haproxy:1.6.4
    labels:
      - "interlock.ext.name=haproxy"
    ports:
      - 80:80
    depends_on:
      - interlock
  web:
    build:
      context: .
      dockerfile: Dockerfile.web
      args:
        - MIX_ENV=dev
    image: schedule
    hostname: docker.dev
    ports:
      - 4000
    environment:
      JWT_SECRET_KEY: |-
        abc123abc123abc123abc123abc123abc123abc123abc123abc123abc
        123abc123abc123abc123abc123abc123abc123abc123abc123abc123
    volumes:
      - ./web:/var/app/web
      - ./config:/var/app/config
      - ./priv:/var/app/priv
      - ./lib:/var/app/lib
      - ./test:/var/app/test
      - ./mix.exs:/var/app/mix.exs
      - ./mix.lock:/var/app/mix.lock
    depends_on:
      - db
  db:
    image: postgres
    environment:
      POSTGRES_PASSWORD: password
    volumes:
      - pg_data:/var/lib/postgresql/data:rw
  registrator:
    image: gliderlabs/registrator:master
    depends_on:
      - consul
    volumes:
      - /var/run/docker.sock:/tmp/docker.sock
    command: -internal -resync 600 consul://consul:8500
  consul:
    image: gliderlabs/consul-server:0.6
    command: "-bootstrap"
  assets:
    build:
      context: .
      dockerfile: Dockerfile.assets
    image: schedule_assets
    volumes:
      - ./web/static:/var/build/web/static
      - ./priv/static:/var/build/priv/static
      - ./.eslintrc:/var/build/.eslintrc
      - ./webpack.config.js:/var/build/webpack.config.js
volumes:
  pg_data:
    driver: local
```

**Comment by princemaple** [comment]
let me know if you need more information.

**Comment by princemaple** [comment]
I'm pretty noob at linux, so...

```
Ubuntu Wily (15.10) (GNU/Linux 4.4.4-std-3 x86_64 )
```

```
 ❯ lshw
scw-10c99b
    description: Computer
    width: 64 bits
    capabilities: smbios-2.8
  *-core
       description: Motherboard
       physical id: 0
     *-memory
          description: System memory
          physical id: 0
          size: 2003MiB
     *-cpu:0
          product: Intel(R) Atom(TM) CPU  C2750  @ 2.40GHz
          vendor: Intel Corp.
          physical id: 1
          bus info: cpu@0
          width: 64 bits
          capabilities: fpu fpu_exception wp vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss syscall nx rdtscp x86-64 constant_tsc arch_perfmon rep_good nopl pni pclmulqdq ssse3 cx16 sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes rdrand hypervisor lahf_lm 3dnowprefetch arat tsc_adjust smep erms
     *-cpu:1
          product: Intel(R) Atom(TM) CPU  C2750  @ 2.40GHz
          vendor: Intel Corp.
          physical id: 2
          bus info: cpu@1
          width: 64 bits
          capabilities: fpu fpu_exception wp vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss syscall nx rdtscp x86-64 constant_tsc arch_perfmon rep_good nopl pni pclmulqdq ssse3 cx16 sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes rdrand hypervisor lahf_lm 3dnowprefetch arat tsc_adjust smep erms
     *-pci
          description: Host bridge
          product: 440FX - 82441FX PMC [Natoma]
          vendor: Intel Corporation
          physical id: 100
          bus info: pci@0000:00:00.0
          version: 02
          width: 32 bits
          clock: 33MHz
        *-isa
             description: ISA bridge
             product: 82371SB PIIX3 ISA [Natoma/Triton II]
             vendor: Intel Corporation
             physical id: 1
             bus info: pci@0000:00:01.0
             version: 00
             width: 32 bits
             clock: 33MHz
             capabilities: isa
             configuration: latency=0
        *-ide UNCLAIMED
             description: IDE interface
             product: 82371SB PIIX3 IDE [Natoma/Triton II]
             vendor: Intel Corporation
             physical id: 1.1
             bus info: pci@0000:00:01.1
             version: 00
             width: 32 bits
             clock: 33MHz
             capabilities: ide
             configuration: latency=0
             resources: ioport:1f0(size=8) ioport:3f6 ioport:170(size=8) ioport:376 ioport:c060(size=16)
        *-bridge UNCLAIMED
             description: Bridge
             product: 82371AB/EB/MB PIIX4 ACPI
             vendor: Intel Corporation
             physical id: 1.3
             bus info: pci@0000:00:01.3
             version: 03
             width: 32 bits
             clock: 33MHz
             capabilities: bridge
             configuration: latency=0
        *-network
             description: Ethernet interface
             product: Virtio network device
             vendor: Red Hat, Inc
             physical id: 2
             bus info: pci@0000:00:02.0
             logical name: eth0
             version: 00
             serial: de:19:44:38:c0:03
             width: 32 bits
             clock: 33MHz
             capabilities: msix bus_master cap_list rom ethernet physical
             configuration: broadcast=yes driver=virtio_net driverversion=1.0.0 ip=10.2.37.199 latency=0 link=yes multicast=yes
             resources: irq:0 ioport:c040(size=32) memory:febc0000-febc0fff memory:feb80000-febbffff
        *-scsi
             description: SCSI storage controller
             product: Virtio block device
             vendor: Red Hat, Inc
             physical id: 3
             bus info: pci@0000:00:03.0
             version: 00
             width: 32 bits
             clock: 33MHz
             capabilities: scsi msix bus_master cap_list
             configuration: driver=virtio-pci latency=0
             resources: irq:0 ioport:c000(size=64) memory:febc1000-febc1fff
```

**Comment by ehazlett** [comment]
Can you give me `docker info` and `docker version` as well?  Thx!

**Comment by ehazlett** [comment]
Also can you run a container and give me the output of `cat /proc/self/cgroup`?  Thx

**Comment by princemaple** [comment]
Docker version is 1.10.3 iirc.
I have to get the rest later. Bed time for Australia.

**Comment by princemaple** [comment]
@ehazlett 

`docker info`

```
 ❯ docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 36
Server Version: 1.10.3
Storage Driver: devicemapper
 Pool Name: docker-254:0-395388-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: ext4
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 2.186 GB
 Data Space Total: 107.4 GB
 Data Space Available: 44.87 GB
 Metadata Space Used: 4.428 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.143 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.99 (2015-06-20)
Execution Driver: native-0.2
Logging Driver: json-file
Plugins:
 Volume: local
 Network: bridge null host
Kernel Version: 4.4.4-std-3
Operating System: Ubuntu 15.10
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.957 GiB
Name: scw-10c99b
ID: 7KER:LGUB:4W3L:D7IK:6C6O:BQLK:KGUG:HJF6:3YYN:YL7Q:ZR6G:3R6S
WARNING: No cpu cfs quota support
WARNING: No cpu cfs period support
```

**Comment by princemaple** [comment]
```
 ❯ docker run --rm nginx:1.9.14 cat /proc/self/cgroup
10:pids:/
9:blkio:/docker/588bca30160dec5f6999b4ed143bf612f5152e2676b029728abd819f1b90f027
8:cpuset:/docker/588bca30160dec5f6999b4ed143bf612f5152e2676b029728abd819f1b90f027
7:freezer:/docker/588bca30160dec5f6999b4ed143bf612f5152e2676b029728abd819f1b90f027
6:net_cls:/docker/588bca30160dec5f6999b4ed143bf612f5152e2676b029728abd819f1b90f027
5:memory:/docker/588bca30160dec5f6999b4ed143bf612f5152e2676b029728abd819f1b90f027
4:devices:/docker/588bca30160dec5f6999b4ed143bf612f5152e2676b029728abd819f1b90f027
3:perf_event:/docker/588bca30160dec5f6999b4ed143bf612f5152e2676b029728abd819f1b90f027
2:cpu,cpuacct:/docker/588bca30160dec5f6999b4ed143bf612f5152e2676b029728abd819f1b90f027
1:name=systemd:/docker/588bca30160dec5f6999b4ed143bf612f5152e2676b029728abd819f1b90f027
```

Running `docker run --rm --entrypoint sh ehazlett/interlock:1.1.0 -c cat /proc/self/cgroup` doesn't seem to output anything...

**Comment by princemaple** [comment]
Surprise, I upgraded the docker installation on the server to use the latest, just released 1.11, and it's working now.
